src/assets/assetsUtil.ts

```diff
@@ -0,0 +1,36 @@
+import { Collectible, CollectibleMetadata } from './AssetsController';
+
+/**
+ * Compares collectible metadata entries to any collectible entry
+ * We need this method when comparing a new fetched collectible metadata, in case a entry changed to a defined value,
+ * there's a need to update the collectible in state.
+ *
+ * @param collectibleMetadata - Collectible metadata object
+ * @param collectible - Collectible object to compare with
+ * @returns - Whether there are differences
+ */
+export function compareCollectiblesMetadata(
+  newCollectibleMetadata: CollectibleMetadata,
+  collectible: Collectible,
+) {
+  const keys: (keyof CollectibleMetadata)[] = [
+    'image',
```

Comment:
Is there a reason why we are not checking all properties of the CollectibleMetedata?

Reply:
yeah it was originally intended to address this issue https://github.com/MetaMask/metamask-mobile/issues/2264